webapp-django/crashstats/api/tests/test_views.py

```diff
@@ -767,27 +680,6 @@ def mocked_supersearch_get(**params):
         })
         assert response.status_code == 200
 
-    def test_change_certain_exceptions_to_bad_request(self):
-        # It actually doesn't matter so much which service we use
-        # because we're heavily mocking it.
-        # Here we use the SuperSearch model.
-
-        def mocked_supersearch_get(**params):
-            if params.get('product'):
-                raise MissingArgumentError(params['product'])
-            else:
-                raise BadArgumentError('That was a bad thing to do')
-
-        SuperSearch.implementation().get.side_effect = mocked_supersearch_get
-
-        url = reverse('api:model_wrapper', args=('SuperSearch',))
-        response = self.client.get(url)
-        assert response.status_code == 400
-        assert 'That was a bad thing to do' in response.content
-        response = self.client.get(url, {'product': 'foobaz'})
-        assert response.status_code == 400
-        assert 'foobaz' in response.content
```

Comment:
I didn't understand this test and the mocking seemed like it created a mythical universe where up is down and left is right, so I moved it upwards, nixed the mocking, and redid it to verify what happens when required parameters were missing and what happens if `BadArgumentError` is thrown.